src/DoctrineAuditBundle/EventSubscriber/AuditSubscriber.php

```diff
@@ -60,14 +59,18 @@ public function onFlush(OnFlushEventArgs $args): void
             $this->manager->processDeletions($em);
 
             $this->manager->reset();
-        }));
+        });
 
-        if ($this->loggerBackup instanceof SQLLogger) {
+        // Embed the chain into the existing LoggerChain, or create a new chain embed the existing SQLLogger.
+        if ($this->loggerBackup instanceof LoggerChain) {
+            $this->loggerBackup->addLogger($auditLogger);
```

Comment:
`$auditLogger` is added at the end of the chain which is not good, it has to be added at the first position of the chain.

Reply:
I don't see a way of adding the AuditLogger to the beginning of the chain without accessing the other loggers in the LoggerChain. LoggerChain's `loggers` variable has no public accessor. 

Not sure if I'm overlooking something?

Reply:
One alternative is to create a new implementation of LoggerChain that has an accessor method but I don't know the ramifications of doing that.

Reply:
You should be able to access loggers inside the `LoggerChain` instance using reflection but it will also probably have an impact performance wise.

Reply:
I actually took a different approach. I made a new implementation of LoggerChain with all the same methods as LoggerChain + a getter for the Loggers. Then I used that as the SQLLogger to set on the entity manager config inside the Audit Subscriber. I will commit it so you can review it and let me know what you think. I think it should be more performant than reflection, and it allows us to put the new AuditLogger in the front of the chain.

Reply:
Looks good to me :)
Just have a look at my last comments and I think it will be all good